File: src/main/java/no/nav/arbeidsgiver/sykefravarsstatistikk/api/felles/OverordnetEnhet.java
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.felles;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import org.jetbrains.annotations.NotNull;

@Data
@Builder
@AllArgsConstructor
public class OverordnetEnhet implements Virksomhet {
    @NotNull
    private final Orgnr orgnr;
    @NotNull
    private final String navn;
    @NotNull
    private final Næringskode5Siffer næringskode;
    @NotNull
    private final InstitusjonellSektorkode institusjonellSektorkode;
    private final int antallAnsatte;
}

File: src/main/kotlin/no/nav/arbeidsgiver/sykefravarsstatistikk/api/applikasjon/domenemodeller/felles/OverordnetEnhet.kt
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.applikasjon.domenemodeller.felles

data class OverordnetEnhet(
    override val orgnr: Orgnr,
    override val navn: String,
    override val næringskode: Næringskode,
    val sektor: Sektor?,
    val antallAnsatte: Int? = 0
) : Virksomhet